### PR TITLE
ICU-20119 urename.h update for icu 63

### PR DESCRIPTION
**Pull request by gnrunge**
- Update urename.h
 - Test uconfig.h variations

##### Checklist

- [ ] Issue filed at https://unicode-org.atlassian.net :  ICU-______
- [ ] Update PR title to include Issue number
- [ ] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

File: icu4c/source/common/unicode/urename.h [hunk 1]
```diff
@@ -109,10 +109,20 @@
 #define _UTF32LEData U_ICU_ENTRY_POINT_RENAME(_UTF32LEData)
 #define _UTF7Data U_ICU_ENTRY_POINT_RENAME(_UTF7Data)
 #define _UTF8Data U_ICU_ENTRY_POINT_RENAME(_UTF8Data)
+#define __bss_start U_ICU_ENTRY_POINT_RENAME(__bss_start)
```
Comment (aheninger) [inline icu4c/source/common/unicode/urename.h]: _bss_start is a system symbol, not an ICU one. I don't think it should be renamed.

File: icu4c/source/common/unicode/urename.h [hunk 2]
```diff
@@ -132,6 +142,7 @@
 #define locale_getKeywordsStart U_ICU_ENTRY_POINT_RENAME(locale_getKeywordsStart)
 #define locale_get_default U_ICU_ENTRY_POINT_RENAME(locale_get_default)
 #define locale_set_default U_ICU_ENTRY_POINT_RENAME(locale_set_default)
+#define maps U_ICU_ENTRY_POINT_RENAME(maps)
```
Comment (aheninger) [inline icu4c/source/common/unicode/urename.h]: maps looks suspicious. Not an API.

File: icu4c/source/common/unicode/urename.h [hunk 3]
```diff
@@ -199,6 +210,7 @@
 #define res_load U_ICU_ENTRY_POINT_RENAME(res_load)
 #define res_read U_ICU_ENTRY_POINT_RENAME(res_read)
 #define res_unload U_ICU_ENTRY_POINT_RENAME(res_unload)
+#define sets U_ICU_ENTRY_POINT_RENAME(sets)
```
Comment (aheninger) [inline icu4c/source/common/unicode/urename.h]: sets is suspicious.

File: icu4c/source/common/unicode/urename.h [hunk 4]
```diff
@@ -109,10 +109,20 @@
 #define _UTF32LEData U_ICU_ENTRY_POINT_RENAME(_UTF32LEData)
 #define _UTF7Data U_ICU_ENTRY_POINT_RENAME(_UTF7Data)
 #define _UTF8Data U_ICU_ENTRY_POINT_RENAME(_UTF8Data)
+#define __bss_start U_ICU_ENTRY_POINT_RENAME(__bss_start)
+#define _edata U_ICU_ENTRY_POINT_RENAME(_edata)
+#define _end U_ICU_ENTRY_POINT_RENAME(_end)
```
Comment (aheninger) [inline icu4c/source/common/unicode/urename.h]: short things like _end and _edata aren't ICU API, and shouldn't be getting exported from the ICU libraries or renamed. We should figure out why they're showing up. Anything starting with _ is questionable.

File: icu4c/source/common/rbbitblb.h [hunk 5]
```diff
@@ -15,6 +15,9 @@
 #define RBBITBLB_H
 
 #include "unicode/utypes.h"
+
+#if !UCONFIG_NO_BREAK_ITERATION
```
Comment (aheninger) [inline icu4c/source/common/rbbitblb.h]: It's unclear why this started needing an #if now. It's an internal header that is only used by the RBBI implementation. With UCONFIG_NO_BREAK_ITERATION set, nothing should include it.

File: icu4c/source/i18n/listformatter.cpp [hunk 6]
```diff
@@ -396,6 +398,7 @@ UnicodeString& ListFormatter::format_(
         int32_t &offset,
         FieldPositionHandler* handler,
         UErrorCode& errorCode) const {
+#if !UCONFIG_NO_FORMATTING
```
Comment (aheninger) [inline icu4c/source/i18n/listformatter.cpp]: There is no 'return' statement outside of the if.

File: icu4c/source/common/rbbitblb.h (hunk 5 above)
Comment (gnrunge) [reply icu4c/source/common/rbbitblb.h]: When compiling this header stand-alone compiler error is:
./common/rbbitblb.h:62:36: error: unknown type name 'IntPair'

'IntPair' is defined in common/rbbirb.h and is filtered out when UCONFIG_NO_BREAK_ITERATION is set.

File: icu4c/source/i18n/listformatter.cpp (hunk 6 above)
Comment (gnrunge) [reply icu4c/source/i18n/listformatter.cpp]: Done

File: icu4c/source/common/unicode/urename.h (hunk 1 above)
Comment (gnrunge) [reply icu4c/source/common/unicode/urename.h]: Fixed (this and a few others (as discussed).

**Pull request merged**